src/main/java/net/sf/jabref/importer/ImportMenuItem.java

```diff
@@ -154,23 +153,9 @@ public void run() {
             for (ImportFormatReader.UnknownFormatImport p : imports) {
                 if (p != null) {
                     ParserResult pr = p.parserResult;
-                    if (pr.hasWarnings()) {
-                        if (Globals.prefs
-                                .getBoolean(JabRefPreferences.DISPLAY_KEY_WARNING_DIALOG_AT_STARTUP)
-                                && pr.hasWarnings()) {
-                            String[] wrns = pr.warnings();
-                            StringBuilder wrn = new StringBuilder();
-                            for (int j = 0; j < wrns.length; j++) {
-                                wrn.append(j + 1).append(". ").append(wrns[j])
-                                        .append("\n");
-                            }
-                            if (wrn.length() > 0) {
-                                wrn.deleteCharAt(wrn.length() - 1);
-                            }
-                            JOptionPane.showMessageDialog(frame, wrn.toString(),
-                                    Localization.lang("Warnings"),
-                                    JOptionPane.WARNING_MESSAGE);
-                        }
+                    if (Globals.prefs.getBoolean(JabRefPreferences.DISPLAY_KEY_WARNING_DIALOG_AT_STARTUP)) {
+                        ParserResultWarningDialog prwDialog = new ParserResultWarningDialog(pr, frame);
+                        prwDialog.show();
```

Comment:
I changed it to a static method showing the dialog. Is this a better or worse method than the previous?